src/User/components/Sidebar/Sidebar.jsx

```diff
@@ -1,48 +1,112 @@
-            <Menu.Item key='5'>option5</Menu.Item>
-            <Menu.Item key='6'>option6</Menu.Item>
-            <Menu.Item key='7'>option7</Menu.Item>
-            <Menu.Item key='8'>option8</Menu.Item>
-          </SubMenu>
-
-          <SubMenu key='sub3' title='subnav 3'>
-            <Menu.Item key='9'>option9</Menu.Item>
-            <Menu.Item key='10'>option10</Menu.Item>
-            <Menu.Item key='11'>option11</Menu.Item>
-            <Menu.Item key='12'>option12</Menu.Item>
-          </SubMenu>
-        </Menu>
-      </Sider>
-    </div>
-  );
+  const dispatch = useDispatch();
+  const [activeLvl0, setActiveLvl0] = useState(null);
+  const [activeLvl1, setActiveLvl1] = useState(null);
+
+  const { allProducts } = useSelector((state) => state.products);
+
+  const getCategories = () => {
+    let allCategories = {};
+
+    for (let category of allProducts) {
+      if (!allCategories[category.hierarchicalCategories.lvl0]) {
+        allCategories[category.hierarchicalCategories.lvl0] = {
+          name: category.hierarchicalCategories.lvl0,
+          children: {},
+        };
+      }
+
+      if (!category.hierarchicalCategories.lvl1) continue;
+
+      let categoryLvl1 = category.hierarchicalCategories.lvl1.split("> ")[1];
+
+      allCategories[category.hierarchicalCategories.lvl0].children[
+        categoryLvl1
+      ] = categoryLvl1;
+    }
+
+    return allCategories;
+  };
+
+  const handleGetCategoryLvl0 = (categoryName, e) => {
+    e.preventDefault();
+    setActiveLvl0(categoryName);
+    setActiveLvl1(null);
+    dispatch(changeFilterCategoryLv0(categoryName));
+  };
+
+  const handleGetCategoryLvl1 = (categoryNameLvl0, categoryNameLvl1, e) => {
+    e.preventDefault();
+    setActiveLvl0(categoryNameLvl0);
+    setActiveLvl1(categoryNameLvl1);
+    dispatch(
+      changeFilterCategoryLv1(`${categoryNameLvl0} > ${categoryNameLvl1}`)
+    );
+  };
+
+  const showCategory = () => {
+    let datas = getCategories();
+
+    let dataView = Object.keys(datas).map((data, key) => {
+      console.log(data);
+      return (
+        <div className='sidebar__list' key={key}>
+          <p
+            className={`sidebar__item  ${
+              datas[data].name === activeLvl0 ? "active" : ""
+            }`}
+            onClick={(e) => handleGetCategoryLvl0(datas[data].name, e)}>
+            {datas[data].name}
+          </p>
+          <span className='sidebar__icon'>
+            {datas[data].name === activeLvl0 ? (
+              <DownOutlined />
+            ) : (
+              <RightOutlined />
+            )}
+          </span>
+
+          {Object.keys(datas[data].children).length > 0 && (
+            <div className='sidebar__sub'>
+              {Object.keys(datas[data].children).map((item, key) => {
+                return (
+                  <div
+                    className={`sidebar__text ${
+                      datas[data].children[item] === activeLvl1
+                        ? "subActive"
```

Comment:
```suggestion
                        ? "sub--active"
```
Để tên class theo dạng snake (sub-active) hoặc BEM thay vì camelCase nhé